lib/cli.js

```diff
@@ -145,22 +146,37 @@ var cli = {
 
         } else {
 
-            engine = new CLIEngine(translateOptions(currentOptions));
             debug("Running on " + (text ? "text" : "files"));
 
-            result = text ? engine.executeOnText(text, currentOptions.stdinFilename) : engine.executeOnFiles(files);
+            // disable --fix for piped-in code until we know how to do it correctly
+            if (text && currentOptions.fix) {
+                console.error("The --fix option is not available for piped-in code.");
+                return 1;
+            }
+
+            engine = new CLIEngine(translateOptions(currentOptions));
+
+            report = text ? engine.executeOnText(text, currentOptions.stdinFilename) : engine.executeOnFiles(files);
             if (currentOptions.quiet) {
-                result.results = CLIEngine.getErrorResults(result.results);
+                debug("Quiet mode enabled - filtering out warnings");
+                report.results = CLIEngine.getErrorResults(report.results);
+            }
+
+            if (currentOptions.fix) {
+                debug("Fix mode enabled - applying fixes");
+                report.results.forEach(function(result) {
```

Comment:
I think it would help if we can make this as a static function on `CLIEngine` so that it can used by modules which consume eslint programmatically.